app/scripts/lib/validate.js

```diff
@@ -146,6 +151,15 @@ define(function (require, exports, module) {
       return _.all(_.keys(data), function (key) {
         return isDataValid(data[key], schema[key]);
       });
+    },
+
+    /**
+     * Check whether string is a url. Validates url with or without http(s) and trailing slash.
+     *
+     * http://blog.mattheworiordan.com/post/13174566389/url-regular-expression-for-links-with-or-without
+     */
+    isUrlValid: function isUrlValid (url){
+      return urlRegEx.test(url);
```

Comment:
Incidentally, you can find some more real-life `urn` values here: https://github.com/mozilla/fxa-oauth-server/blob/master/config/dev.json#L44

I'm no expert in this part of the code, but I suspect those tests are using instances of `SyncRelier` out of convenience, and some of them could be refactored to use `OAuthRelier`.  Certainly `OAuthRelier` is the only relier subclass that explicitly lists `redirectUri` among its default fields.